### PR TITLE
Add Heuristics 22 editorial

### PR DESCRIPTION
**Pull request by Uditgulati**

### CODE REVIEW

File: _posts/2018-09-09-heuristics-22-editorial.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+[Solution](https://gist.github.com/Uditgulati/3fbc2374655fe4b8df5c0493a87ade77#file-assigning-heroes-cpp)
+
+**Overall Complexity:** $$O(T * max(N, M))$$
+
+---
+
+## Make 'em One
+
+The problem states that given an array and some value K and a range $$[L, R]$$, find the sum of absolute difference of each element of the range with K. This can be done in $$O(logN)$$ per query using merge sort tree. Construct the merge sort tree of the array and also, store prefix sum of list in the merge sort tree. Use the merge sort tree to get the sum of all values below K and same goes for values above K. Let X be the count of values below K and Y be the count of values above K. Then, answer would be $$(X * K - sum of values below K) + (sum of values above K - Y * K)$$.
```
Comment (paramsingh) [inline _posts/2018-09-09-heuristics-22-editorial.md:49]: There's no merge sort tree involved in the solution you link.

Comment (Uditgulati) [reply _posts/2018-09-09-heuristics-22-editorial.md:49]: Oh.. That's actually the tester's code. Didn't get the time to look at it and assumed that it was same as that of mine. I'll either write the solution given in editorial or ask the tester to add explanation to his solution.

Comment (paramsingh) [reply _posts/2018-09-09-heuristics-22-editorial.md:49]: 😂 Tester is too busy.

Comment (Uditgulati) [reply _posts/2018-09-09-heuristics-22-editorial.md:49]: Lol... So is the author :rofl:. I'll see if this can be done by this weekend.

Comment (Uditgulati) [reply _posts/2018-09-09-heuristics-22-editorial.md:49]: Let's do this, merge this as it is and open an issue to write a code that matches editorial or vise-versa. Maybe someone kind enough will make a PR for it. :laughing:

**Pull request merged**